File: frontend angular/src/app/interface/question.ts
import { logging } from 'protractor';
import { ICategory } from './category';
import { IUser } from './user';
import { IAnswer } from './answer';

export class IQuestion{
    public id: number;
    public user:IUser;
    public answers: Array<IAnswer>;
    public title: string;
    public description: string;
    public category: ICategory;

     constructor(){}

    //  setTitle(title: string){
    //      this.title=title;
    //  }

    //  setDescription(description: string){
    //     this.description=description;
    // }

    // setCatId(catId: number){
    //     this.category.setId(catId);
    // }


}

File: api/src/main/java/com/askme/askme/models/Question.java
package com.askme.askme.models;

import com.fasterxml.jackson.annotation.JsonIgnore;
import org.springframework.web.bind.annotation.RequestParam;

import javax.persistence.*;
import java.util.List;

@Entity
@Table(name = "question")
public class Question {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(length = 500)
    private String title;

    @Column(length = 2000)
    private String description;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    @OneToMany(mappedBy = "question", fetch = FetchType.LAZY)
    @JsonIgnore
    private List<Answer> answers;

    @ManyToOne
    @JoinColumn(name = "c_id")
    private Category category;

    public Question() {
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public Category getCategory() {
        return category;
    }

    public void setCategory(Category category) {
        this.category = category;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public List<Answer> getAnswers() {
        return answers;
    }

    public void setAnswers(List<Answer> answers) {
        this.answers = answers;
    }
}
